src/main/kotlin/net/leanix/githubagent/services/CachingService.kt

```diff
@@ -0,0 +1,26 @@
+package net.leanix.githubagent.services
+
+import jakarta.annotation.PostConstruct
+import net.leanix.githubagent.config.GithubEnterpriseProperties
+import org.springframework.stereotype.Service
+
+@Service
+class CachingService(
+    private val githubEnterpriseProperties: GithubEnterpriseProperties
+) {
+    private val cache = HashMap<String, String?>()
```

Comment:
Have you tried one of these caches here? https://docs.spring.io/spring-boot/reference/io/caching.html

Reply:
Is this really a cache? I mean, does the cached values eventually expire? Maybe the naming is a bit confusing.

Reply:
Ok, got it now, you just need to save somewhere the token.